stonesoup/initiator/tests/test_simple.py

```diff
@@ -22,19 +23,24 @@
 )
 
 
-def test_spi():
+@pytest.mark.parametrize(
+    'measurement_model',
+    [LinearGaussian(2, [0, 1], np.diag([1, 1])),
+     CartesianToBearingRange(2, [1, 0], np.diag([0.1, 1]))],
+    ids=['linear', 'non-linear'])
+def test_spi(measurement_model):
     """Test SinglePointInitiator"""
 
     # Prior state information
     prior_state = GaussianState(
         np.array([[0], [0]]),
         np.array([[100, 0], [0, 1]]))
 
-    # Define a measurement model
-    measurement_model = LinearGaussian(2, [0], np.array([[1]]))
-
     # Create the Kalman updater
-    kup = KalmanUpdater(measurement_model)
+    if isinstance(measurement_model, LinearModel):
```

Comment:
You don't really need this though do you? In event of linear model `ExtendedKalmanUpdater` uses `KalmanUpdater` logic.

Reply:
True, but as the test uses the `KalmanUpdater` and initiator uses `ExtendedKalmanUpdater` it helps to test that the output is the same and that is indeed True.

Reply:
Ah, well done, +1 for test within a test.